### PR TITLE
Add support for reading and writing the device description to/from flash

### PR DESCRIPTION
**Pull request by ademuri**

### CODE REVIEW

File: lib/device/DeviceDescription.hpp [hunk 1]
```diff
@@ -22,5 +28,11 @@ class DeviceDescription {
                              const std::vector<StripDescription> strips);
 
   uint8_t GetLedCount() const;
+
+  static constexpr uint32_t kCheckValue = 0x12345678;
+
+  uint32_t check_value = kCheckValue;
+
+  static constexpr uint32_t kMaxSize = 128;
```
Comment (wmmiii) [inline lib/device/DeviceDescription.hpp:36]: This is the number of bytes reserved for the device description to fit in?

Comment (ademuri) [reply lib/device/DeviceDescription.hpp:36]: Correct. There's a compile-time assertion in `Devices.hpp` that the current device fits into this limit.

File: src/devices/node/node.cpp [hunk 2]
```diff
@@ -14,26 +16,86 @@
...
-FastLedManager led_manager(Devices::current, &state_machine);
+FastLedManager *led_manager;
+
+constexpr uint32_t kEepromStart = 0x00020000 - 0x2000;
+FlashClass device_storage(/*flash_addr=*/(void *)kEepromStart,
+                          /*size=*/DeviceDescription::kMaxSize);
 
 void FeedWatchdog() { WDT->CLEAR.reg = WDT_CLEAR_CLEAR_KEY; }
 
+FastLedManager *ReadDeviceFromFlash() {
```
Comment (wmmiii) [inline src/devices/node/node.cpp:34]: This appears to be repeated, can we DRY the function up?

**Comment by wmmiii** [comment]
Could you create a short README on how to use this functionality?

File: src/devices/node/node.cpp (hunk 2 above)
Comment (ademuri) [reply src/devices/node/node.cpp:34]: The APIs for flash storage between the two platforms are just different enough that I think it makes sense to have this repeated.

**Comment by ademuri** [comment]
Added documentation to the readme.

**Pull request merged**